muikku-core-plugins/src/main/resources/META-INF/resources/scripts/src/components/general/form-element.tsx

```diff
@@ -146,37 +145,59 @@ export class FormWizardActions extends React.Component<FormWizardActionsProps, F
 
 interface SearchFormElementProps {
   updateField: (value: string) => any,
-  value?: string,
+  value: string,
+  id: string,
   name: string,
-  placeholder?: string,
-  modifiers?: string | Array<string>
+  onFocus?: () => any,
+  onBlur?: () => any,
+  placeholder: string,
+  modifiers?: string | Array<string>,
+  className?: string,
 }
 
 interface SearchFormElementState {
-  value: string
+  clearActive: boolean
 }
 
 export class SearchFormElement extends React.Component<SearchFormElementProps, SearchFormElementState> {
 
+  private searchInput: React.RefObject<HTMLInputElement>;
+
   constructor(props: SearchFormElementProps) {
     super(props);
+
+    this.updateSearchField = this.updateSearchField.bind(this);
+    this.clearSearchField = this.clearSearchField.bind(this);
+    this.searchInput = React.createRef();
+
     this.state = {
-      value: this.props.value
+      clearActive: false
```

Comment:
What's the purpose of clearActive? I find no purpose for it, could we remove it?....